### DISCUSSION TITLE
Deleted deployments still reachable

### DISCUSSION THREAD

**Discussion by tresabhi** [vercel/community#7434] [Help] [answered]
### Summary

I enabled monetization on my production site so I have started deleting preview deployments so that all traffic goes to the main site. I used the command `vercel delete [project] --safe` and it seems to have deleted a few deployments before stopping due to reaching the rate limit:

![Screenshot_20240711-092950_Brave](https://github.com/vercel/community/assets/43380238/ea7ba593-5ff8-4126-9762-42ace22054e9)

I ran this command a few times every 10 minutes as suggested by the rate limits but the command seems to be lying to me now? In the image above, you can see that the deployments have indeed been deleted but navigating to those deployment URLs shows that they are still up and running. First url in image: blitzkrieg-caqhx6h2s-tresabhi.vercel.app

### Steps to Reproduce

1. Start with a massive project (mine has like 1.5K commits/preview deployments)
2. Run the `vercel delete [project] --safe` command
3. Watch it delete some and then fail due to rate limits
4. Try again in 10 minutes over and over again
5. Go to the Vercel dashboard activities tab
6. Note that it does say that many deployments have been deleted
7. Copy the link to the allegedly deleted deployments and go to the URLs
8. Frown at the deployments still being up

**Comment by pawlean** [comment]
Hi @tresabhi!

That is odd, I checked on ["What's my DNS"](https://www.whatsmydns.net/#CNAME/blitzkrieg-caqhx6h2s-tresabhi.vercel.app) and it looks like it's not working.

Sometimes Deleted URLs may still be reachable due to DNS propagation delays.

But I can access it... Strange! I will investigate and get back to you.

**Comment by tresabhi** [comment]
Thank you for the hasty reply. For what it's worth, according to the CLI, now all of the `--safe` deployments have been deleted. And of course, most of them are reachable.
![](https://i.imgur.com/uxpi12v.png)

**Comment by tresabhi** [comment]
@pawlean any updates?

**Reply by pawlean** [reply]
Hi @tresabhi! Do you still see this? From what I can see (still have to confirm this internally), but these get deleted after 14 days.

**Reply by tresabhi** [reply]
They're still up

**Reply by pawlean** [reply]
Checking internally. Thanks for the patience!

**Reply by pawlean** [reply]
@tresabhi Deployments are hard-deleted here after 30 days. Can you let me know if you still see this after 30 days?

**Reply by tresabhi** [reply]
You got it

**Reply by tresabhi** [reply]
It indeed does! Closing.

**Comment by leerob** [comment, marked as answer]
You can now configure the retention period for deployments through the dashboard and CLI.

For example, canceled and errored deployments might be set to 30 days retention, while production deployments might be set to 1 year. Recently deleted deployments are shown in your project settings and can be instantly restored within 30 days of deletion.

Learn more in our [documentation](https://vercel.com/docs/security/deployment-retention).

**Reply by tresabhi** [reply]
![image](https://github.com/user-attachments/assets/15f3e0d1-b04f-47ae-aef1-d645e7aab2c0)

Fonts are a little messed up but glad to see this!

[a comment by an automated account is not shown]
